test/decoder.js

```diff
@@ -60,6 +60,23 @@ test('decode wiki example', function(done) {
 
     decoder.write(msg.replace(/\|/g, sep));
 });
+test('Filled up buffer', function(done) {
+    var decoder = FrameDecoder();
+    String.prototype.repeat = function (num) {
```

Comment:
can this be avoided? Prefer not to modify the string prototype (I know it is in tests but still).